### ISSUE TITLE
Feature request: Show when you have a new record

### ISSUE THREAD

**Issue by RustanHakansson** [lexica/lexica#167]
It would be great to show when you have a new record. A reward, a small extra reason to push to do better. Even in the very first game, to immediately get an incentive to try again. 

During the game, show the text "New record!" in small text somewhere, perhaps just above the current score. 

But especially after the game is over, preferably with a small splash screen overlaid on the end screen. A simple celebration animation, or just words showing the old record and the new record, and the game mode. Additional messages to display could be how long time since the old record was set.

**Comment by pserwylo** [comment]
Yes, 100%. I want to make the game have a little bit of flair about it. A high score screen is a very good first step. Indeed there is currently no way to view past high scores. 

It is a little bit tied to the "Game Mode" notion that we've been discussing, so I think I'd like to make a move on that sooner rather than later.

This could be built in parallel though, because it is a separate activity which needs to be plugged into the post-game section regardless of whether we support game modes or not.

I've started documenting this in the wiki (referencing back to this issue) because I think sometimes that can be a better place to post long term plans: https://github.com/lexica/lexica/wiki#high-score-tables.

**Comment by RustanHakansson** [comment]
Maybe let the user choose how highscores are counted, either as total words and total score, which varies a lot currently as it is strongly tied to the number of possible words (easier to get high scores if there are 200 words than if there are 40), or if they want to do this as a percentage of the possible score.

**Comment by dirby7** [comment]
Also, track high score separately for each game variant, such as time limit or min word len, etc.
